### PR TITLE
feat(.NET): Update migration guide for `4.x`

### PR DESCRIPTION
**Pull request by bitsandfoxes**
Fixes https://github.com/getsentry/sentry-dotnet/issues/2673

### CODE REVIEW

File: docs/platforms/dotnet/migration/index.mdx [hunk 1]
```diff
@@ -6,6 +6,138 @@ sidebar_order: 8000
...
+- A number of `[Obsolete]` options have been removed
+  - `BeforeSend` - use `SetBeforeSend` instead.
+  - `BeforeSendTransaction` - use `SetBeforeSendTransaction` instead.
+  - `BeforeBreadcrumb` - use `SetBeforeBreadcrumb` instead.
+  - `CreateHttpClientHandler` - use `CreateHttpMessageHandler` instead.
+  - `ReportAssemblies` - use `ReportAssembliesMode` instead.
+  - `KeepAggregateException` - this property is no longer used and has no replacement.
+  - `DisableTaskUnobservedTaskExceptionCapture` method has been renamed to `DisableUnobservedTaskExceptionCapture`.
+  - `DebugDiagnosticLogger` - use `TraceDiagnosticLogger` instead.
+- A number of iOS/Android-specific `[Obsolete]` options have been removed
```
Comment (vivianyentran) [inline docs/platforms/dotnet/migration/index.mdx:93]:
```suggestion
- A number of iOS/Android-specific `[Obsolete]` options have been removed:
```

File: docs/platforms/dotnet/migration/index.mdx [hunk 2]
```diff
@@ -6,6 +6,138 @@ sidebar_order: 8000
...
+  - `StitchAsyncCode` - no longer available.
+  - `ProfilingTracesInterval` - no longer available.
+  - `ProfilingEnabled` - use `ProfilesSampleRate` instead.
+- Obsolete `SystemClock` constructor removed, use `SystemClock.Clock` instead.
+- Obsolete `Runtime.Clone()` removed, this shouldn't have been public in the past and has no replacement.
+- Obsolete `SentryException.Data` removed, use `SentryException.Mechanism.Data` instead.
+- Obsolete `AssemblyExtensions` removed, this shouldn't have been public in the past and has no replacement.
+- Obsolete `SentryDatabaseLogging.UseBreadcrumbs()` removed, it is called automatically and has no replacement.
+- Obsolete `Scope.GetSpan()` removed, use `Span` property instead.
+- Obsolete `IUserFactory` removed, use `ISentryUserFactory` instead.
```
Comment (vivianyentran) [inline docs/platforms/dotnet/migration/index.mdx:109]:
```suggestion
- Obsolete `SystemClock` constructor removed. Use `SystemClock.Clock` instead.
- Obsolete `Runtime.Clone()` removed. This shouldn't have been public in the past and has no replacement.
- Obsolete `SentryException.Data` removed. Use `SentryException.Mechanism.Data` instead.
- Obsolete `AssemblyExtensions` removed. This shouldn't have been public in the past and has no replacement.
- Obsolete `SentryDatabaseLogging.UseBreadcrumbs()` removed. It is called automatically and has no replacement.
- Obsolete `Scope.GetSpan()` removed. Use `Span` property instead.
- Obsolete `IUserFactory` removed. Use `ISentryUserFactory` instead.
```

File: docs/platforms/dotnet/migration/index.mdx [hunk 3]
```diff
@@ -6,6 +6,138 @@ sidebar_order: 8000
...
+- `IHasMeasurements` has been removed, use `ISpanData` instead.
+- `IHasBreadcrumbs` has been removed, use `IEventLike` instead.
+- `ISpanContext` has been removed, use `ITraceContext` instead.
+- `IHasTransactionNameSource` has been removed, use `ITransactionContext` instead.
```
Comment (vivianyentran) [inline docs/platforms/dotnet/migration/index.mdx:113]:
```suggestion
- `IHasMeasurements` has been removed. Use `ISpanData` instead.
- `IHasBreadcrumbs` has been removed. Use `IEventLike` instead.
- `ISpanContext` has been removed. Use `ITraceContext` instead.
- `IHasTransactionNameSource` has been removed. Use `ITransactionContext` instead.
```

File: docs/platforms/dotnet/migration/index.mdx [hunk 4]
```diff
@@ -6,6 +6,138 @@ sidebar_order: 8000
...
+- The unused `StackFrame.InstructionOffset` has been removed.
+- The unused `Scope.Platform` property has been removed.
+- The obsolete setter `Sentry.PlatformAbstractions.Runtime.Identifier` has been removed
```
Comment (vivianyentran) [inline docs/platforms/dotnet/migration/index.mdx:116]:
```suggestion
- The obsolete setter `Sentry.PlatformAbstractions.Runtime.Identifier` has been removed.
```

File: docs/platforms/dotnet/migration/index.mdx [hunk 5]
```diff
@@ -6,6 +6,138 @@ sidebar_order: 8000
...
+- `Sentry.Values<T>` is now internal as it is never exposed in the public API
```
Comment (vivianyentran) [inline docs/platforms/dotnet/migration/index.mdx:117]:
```suggestion
- `Sentry.Values<T>` is now internal as it is never exposed in the public API.
```

File: docs/platforms/dotnet/migration/index.mdx [hunk 6]
```diff
@@ -6,6 +6,138 @@ sidebar_order: 8000
...
+- The `TracePropagationTarget` class has been removed, use the `SubstringOrRegexPattern` class instead.
```
Comment (vivianyentran) [inline docs/platforms/dotnet/migration/index.mdx:118]:
```suggestion
- The `TracePropagationTarget` class has been removed. Use the `SubstringOrRegexPattern` class instead.
```

File: docs/platforms/dotnet/migration/index.mdx (hunk 2 above)
Comment (vivianyentran) [reply docs/platforms/dotnet/migration/index.mdx:109]: Is there a good way to group these and the bullets below it together somehow? like how the iOS/Android options are together

**Review by vivianyentran — APPROVED** [review]
Added some suggestions

File: docs/platforms/dotnet/migration/index.mdx [hunk 7]
```diff
@@ -6,6 +6,138 @@ sidebar_order: 8000
...
+
+We're dropping support for some of the old target frameworks, please check this [GitHub Discussion](https://github.com/getsentry/sentry-dotnet/discussions/2776) for details on why.
+
+- Support for .NET Framework 4.6.1 has been replaced with 4.6.2 
+<br/>.NET Framework 4.6.1 was announced on Nov 30, 2015. And went out of support over a year ago, on Apr 26, 2022.
+- Support for .NET Core 3.1 and .NET 5 has been dropped
+- Support for netstandard2.0 for Sentry.AspNetCore has been dropped
+- Support for .NET 6 on mobile (e.g: `net6.0-android`) has been replaced with .NET 7 
+<br/>.NET 6 on mobile has been out of support since May 2023 and with .NET 8, it's no longer possible to build .NET 6 Mobile specific targets. For that reason, we're moving the mobile-specific TFMs from `net6.0-platform` to `net7.0-platform`. Mobile apps still work on .NET 6 will pull the `Sentry` .NET 6, which offers the .NET-only features, without native/platform-specific bindings and SDKs. See [this ticket for more details](https://github.com/getsentry/sentry-dotnet/issues/2623).
+- Support for MAUI Tizen has been dropped
```
[the remark this answers is not in the document]
Comment (bitsandfoxes) [reply docs/platforms/dotnet/migration/index.mdx:20]: I updated that part.

**Pull request merged**